### PR TITLE
kraken: qa/suites: do not upgrade ceph-{base,mon,osd} on client side in upgra…

### PR DESCRIPTION
**Pull request by tchaikov**
…de tests

Fixes: http://tracker.ceph.com/issues/22558
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

File: qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ tasks:
     exclude_packages: ['ceph-mgr','libcephfs2','libcephfs-devel','libcephfs-dev']
 - print: "**** done install jewel"
```
Comment (smithfarm) [inline qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml]: should we do s/jewel/kraken/ here?

File: qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ tasks:
...
 - install.upgrade:
-   exclude_packages: ['ceph-test', 'ceph-test-dbg','libcephfs1']
+   exclude_packages: ['ceph-base', 'ceph-mon', 'ceph-osd', 'ceph-test', 'ceph-test-dbg','libcephfs1']
    client.1:
 - print: "**** done install.upgrade to -x on client.0"
```
Comment (smithfarm) [inline qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml]: should we do s/client.0/client.1/ here?

File: qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml (hunk 1 above)
Comment (tchaikov) [reply qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml]: actually it depends if we are running this test against jewel, it's jewel. if we are running this test against luminous, it'd be luminous.

File: qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml (hunk 2 above)
Comment (smithfarm) [reply qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml]: judging from the yum command [1] in the log, the packages list does not contain ceph-{base,mon,osd} explicitly, but rather just "ceph". So I wonder if this exclude will work.

[1] https://tracker.ceph.com/issues/22558#note-11

Comment (tchaikov) [reply qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml]: it's not in the scope of this change. but if you insist, i will add another commit in this PR.

File: qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml (hunk 1 above)
Comment (smithfarm) [reply qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml]: even when the install block here says `branch: kraken`?

File: qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml (hunk 2 above)
Comment (smithfarm) [reply qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml]: I could open another PR to fix the documentation strings, if you insist :-P

Comment (tchaikov) [reply qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml]: @smithfarm agreed, it won't work. 

- the ceph rpm package will bring in ceph-{osd,mds,mgr,mon} on , and
- the ceph deb package will bring in ceph-{mgr,mon,osd}.

i will also list "ceph" in the exclude list.

Comment (smithfarm) [reply qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml]: whoa, I just noticed: `sudo yum -y install '' libcephfs_jni1 rbd-fuse ceph-radosgw librbd1 ceph-fuse python-ceph f ceph-mgr ceph librados2 cephfs-java libcephfs2 libcephfs-devel`

There appears to be a package `f` specified. . . that's weird.

File: qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml (hunk 1 above)
Comment (tchaikov) [reply qa/suites/upgrade/client-upgrade-kraken/kraken-client-x/rbd/1-install/kraken-client-x.yaml]: oh, that's right.

**Comment by tchaikov** [comment]
@smithfarm updated. please note, i am just trying to fix the test failure. so i intended to have a minimal fix.

**Comment by tchaikov** [comment]
> It's a little worrying that we are now excluding `ceph-common` from the packages to upgrade on the client, but for this test case the main thing is to upgrade the client libraries.

i don't see what is worrying you. `ceph-common` is not directly related to whether or not `librados` is installed. and the latter is always installed if not excluded by the list, see https://github.com/ceph/teuthology/blob/master/teuthology/task/install/packages.yaml#L33

**Comment by yuriw** [comment]
test this please

**Comment by smithfarm** [comment]
@tchaikov https://github.com/ceph/ceph/pull/20413 was merged so I'm closing this one - reopen if you don't agree.

**Pull request closed without merging**